utils/frame/benchmarking-cli/src/pallet/command.rs

```diff
@@ -508,20 +509,31 @@ impl PalletCmd {
 			self.print_summary(&batches, &storage_info, pov_modes.clone())
 		}
 
-		// Create the weights.rs file.
+		// Write all weight files and possibly error afterwards.
+		let mut soft_errors = Vec::new();
 		if let Some(output_path) = &self.output {
-			writer::write_results(
+			if let Err(err) = writer::write_results(
 				&batches,
 				&storage_info,
 				&component_ranges,
 				pov_modes,
 				self.default_pov_mode,
 				output_path,
 				self,
-			)?;
+			)? {
```

Comment:
This is the trick :yum: I originally put a comment here, but the formatter moved it down and so it was pointless.